pkg/mermaid/datamodeldiagram/datamodeldiagramwithappandtype.go

```diff
@@ -0,0 +1,81 @@
+package datamodeldiagram
+
+import (
+	"fmt"
+
+	"github.com/anz-bank/sysl/pkg/mermaid"
+	"github.com/anz-bank/sysl/pkg/sysl"
+	"github.com/anz-bank/sysl/pkg/syslwrapper"
+)
+
+func GenerateDataModelDiagramWithAppAndType(m *sysl.Module, appName string, typeName string) (string, error) {
+	mapper := syslwrapper.MakeAppMapper(m)
+	mapper.IndexTypes()
+	mapper.ConvertTypes()
+	return generateDataModelDiagramWithAppAndTypeHelper(mapper.SimpleTypes, appName, typeName,
+		&[]externalLink{}, &[]string{}, &[]externalLink{})
+}
+
+func generateDataModelDiagramWithAppAndTypeHelper(simpleTypes map[string]*syslwrapper.Type, appName string,
+	typeName string, externalLinks *[]externalLink, appTypes *[]string, eLinks *[]externalLink) (string, error) {
+	var result string
+	appType1 := fmt.Sprintf("%s:%s", appName, typeName)
+	result = "%% AUTOGENERATED CODE -- DO NOT EDIT!\n\nclassDiagram\n"
```

Comment:
`%% AUTOGENERATED CODE -- DO NOT EDIT!\n\nclassDiagram\n` is also defined in datamodeldiagram.go, perhaps it'd be defined as a single constant, that can be shared?

I think it'd also be cleaner to give them names like so
```
const generatedHeader =""%% AUTOGENERATED CODE -- DO NOT EDIT!\n" 
const classDiagramHeader ="\nclassDiagram\n"
```

Reply:
Addressed!

Reply:
the class diagram header isn't repeated, hence I haven't changed it.